src/vs/workbench/browser/web.api.ts

```diff
@@ -578,6 +583,34 @@ export interface IInitialColorTheme {
 	readonly colors?: { [colorId: string]: string };
 }
 
+export interface IWelcomeDialog {
+
+	/**
+	 *  Unique identifier of the vscode.dev route constructing the welcome dialog.
+	 */
+	routeId: string;
```

Comment:
Thinking about this more, perhaps it should be renamed to `id` to be more generic across consumers, since the web api is consumed by others besides vscode.dev

Reply:
But what is this used for actually?

Reply:
I'm currently using this to check if a dialog from this route has been previously shown. I foresee its use evolving with the welcome dialog.